Sources/BowEffects/Data/IO.swift

```diff
@@ -284,17 +284,10 @@ private class AsyncIO<E: Error, A> : IO<E, A> {
     
     override func attempt() -> IO<E, A> {
         var result: IO<E, A>?
-        
-        do {
-            let callback: Callback<E, A> = { either in
-                result = either.fold(RaiseError.init, Pure.init)
-            }
-            try f(callback)
-        } catch let error as E {
-            result = RaiseError<E, A>(error)
-        } catch {
-            fatalError("IO did not handle error: \(error). Only errors of type \(E.self) are handled.")
+        let callback: Callback<E, A> = { either in
+            result = either.fold(RaiseError.init, Pure.init)
         }
+        f(callback)
         
         while(result == nil) {}
```

Comment:
Yes, the ideal would be a `DispatchGroup`. Since `IO` will be refactored to conform to the new type class hierarchy, I'd leave this out for the moment. Issue #285 tracks this.

Reply:
Great!